go/vt/vtgate/semantics/early_rewriter.go

```diff
@@ -138,6 +140,45 @@ func rewriteOrExpr(cursor *sqlparser.Cursor, node *sqlparser.OrExpr) {
 	}
 }
 
+// rewriteAndExpr rewrites AND expressions when either side is TRUE.
+func rewriteAndExpr(cursor *sqlparser.Cursor, node *sqlparser.AndExpr) {
+	newNode := rewriteAndTrue(*node)
+	if newNode != nil {
+		cursor.ReplaceAndRevisit(newNode)
+	}
+}
+
+func rewriteAndTrue(andExpr sqlparser.AndExpr) sqlparser.Expr {
+	// we are looking for the pattern `WHERE c = 1 AND 1 = 1`
+	isTrue := func(subExpr sqlparser.Expr) bool {
+		evalEnginePred, err := evalengine.Translate(subExpr, nil)
+		if err != nil {
+			return false
+		}
+
+		env := evalengine.EmptyExpressionEnv()
+		res, err := env.Evaluate(evalEnginePred)
+		if err != nil {
```

Comment:
constant folding is getting implemented in evalengine. This might not be needed later.

Reply:
Maybe, we have the equivalent of this for ORs so I added it for ANDs too. For now this just makes our plans not have redundant operations like 1 = 1.